zerver/tests/test_signup.py

```diff
@@ -1610,6 +1610,38 @@ def test_accessing_invites_in_another_realm(self) -> None:
         error_result = self.client_delete('/json/invites/' + str(prereg_user.id))
         self.assert_json_error(error_result, "No such invitation")
 
+    def test_prereg_user_status(self) -> None:
+        email = self.nonreg_email("alice")
+        password = "password"
+        realm = get_realm('zulip')
+
+        inviter = UserProfile.objects.first()
+        prereg_user = PreregistrationUser.objects.create(
+            email=email, referred_by=inviter, realm=realm)
+
+        confirmation_link = create_confirmation_link(prereg_user, 'host', Confirmation.USER_REGISTRATION)
+        registration_key = confirmation_link.split('/')[-1]
+
+        confirmation = Confirmation.objects.get(confirmation_key=registration_key)
+        prereg_user = confirmation.content_object
+
+        result = self.client_post(
+            "/accounts/register/",
+            {"key": registration_key,
+             "from_confirmation": "1",
+             "full_name": "alice"})
+        self.assertEqual(prereg_user.status, 0)
```

Comment:
I don't think this does what you think it does -- we're checking the attributes of this object that we fetched from the database before making the change that might have improperly actived the link.

`prereg_user.refresh_from_db()` on the line before this might be helpful, but probably better would be to just do the `Confirmation` query etc. after the `client_post`.